src/settings.cpp

```diff
@@ -1899,13 +1899,14 @@ bool SetSettingValue(const StringSettingDesc *sd, std::string value, bool force_
  * @param object The object the setting is in.
  * @param newval The new value for the setting.
  */
-void StringSettingDesc::ChangeValue(const void *object, std::string &newval) const
+void StringSettingDesc::ChangeValue(const void *object, std::string_view newval) const
```

Comment:
Weird old code, but the caller actually does no longer use the string.
```suggestion
void StringSettingDesc::ChangeValue(const void *object, std::string &&newval) const
```